### PR TITLE
WIP: Use commands programmatically

### PR DESCRIPTION
**Pull request by aurelienshz**
## Purpose:
Make all CLI features usable prgrammatically, in a Node.js script. This helps for some advanced use cases where the users want to be able to script the upload process, without making the CLI more complex. For example #38 could be solved by having an upload script responsible for reading the environment config files, and write the correct config values before calling clasp. This also allows to have Clasp integrate nicely with a build system such as Gulp.

Another nice side effect is that this PR makes testing possible (since we can import the features from a test file).

## Updates performed:

* Make the build script of this tool capable of handling several files inside a src directory
* Import these features in cli.ts, where all things CLI-related have been moved (ie all commander calls)

### CODE REVIEW

**Comment by grant** [comment]
@aurelienshz Thanks for the PR.
I don't think I can approve of a PR that modifies the whole library without having a large discussion and design doc around it.
### Do you mind
- Discussing what issue you face
- Propose a small PR that fixes that issue?

I think we should this PR in favor of discussing first and making small PRs.

**Comment by aurelienshz** [comment]
Hi @grant !

You are perfectly right. I started refactoring and got completely carried away, and started over-refactoring and writing the PR draft alongside with random ideas that came to me. Sorry :blush: 

I have just updated the description and pushed commits on the branch that make the scope of the PR a lot clearer, and greatly reduce the changeset.

File: src/cli.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+import {
+  PROJECT_NAME, ERROR, LoginOptions,
+  login, logout, create, clone, pull, push, openScriptProject,
+  listDeployments, deploy, undeploy, redeploy, listVersions, createVersion
+} from "./index";
+
+/**
+ * Set global CLI configurations
+ */
+commander
```
Comment (grant) [inline src/cli.ts]: Laying out `commander` commands like this is nice.
I'd support this change.

**Comment by grant** [comment]
I appreciate the enthusiasm, but I don't think there are many advantages of using clasp as an API. There's the Apps Script API for that, and it's not complex.

This PR is changing too much. If you want to modify this PR to **only** move the `command` commands together (like I commented) and not export the commands, I'd approve of that. 

Otherwise, I'm in favor of closing this and discussing more in GitHub issues.

**Comment by aurelienshz** [comment]
Closed in favor of #68

**Comment by grant** [comment]
If we want to create an API, we need smaller PRs and more discussion about design and code quality.
I don't see why you can't build an API that calls the CLI if that's the goal.

**Pull request closed without merging**